0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,662 +116,889 @@ informative:
+Note that the example given, while useful to demonstrate the structure, has less practical security value that the use of an RSA-SHA-256 or ED25519-SHA-256 subfulfillment. Since the subfulfillment is a PREIMAGE-SHA-256, the validation of which ignores the incoming message, as long as the prefix, maxMessagelength and preimage provided in the subfulfillment are correct, the parent PREFIX-SHA-256 fulfillment will validate.
 
-threshold
-: threshold MUST be an integer in the range 1 ... 2^32 - 1. In order to fulfill a threshold condition, the weights of the provided fulfillments MUST be greater than or equal to the threshold.
+In this case, wrapping the PREIMAGE-SHA-256 crypto-condition in the PREFIX-SHA-256 crypto-condition, has the effect of enforcing a message length of 0 bytes.
 
-subconditions
-: is the set of subconditions, each provided as a tuple of weight and condition.
+Note also, any change to the PREFIX-SHA-256 crypto-condition's prefix and maxMessageLength values result in a different fingerprint value, effectively namespacing the underlying preimage and re-hashing it. The result is a new crypto-condition with a new and unique fingerprint with no change to the underlying sub-crypto-condition.
 
-weight
-: is the numeric weight of this subcondition, i.e. how many times it counts against the threshold.
+## THRESHOLD-SHA-256 {#threshold-sha-256-condition-type}
+THRESHOLD-SHA-256 is assigned the type ID 2. It relies on the availability of the SHA-256 digest algorithm and any other algorithms required by any of its sub-crypto-conditions as it is a compound crypto-condition type.
+
+### Cost {#threshold-sha-256-condition-type-cost}
 
-condition
-: is the subcondition.
+The cost is the sum of the F.threshold largest cost values of all sub-conditions, added to 1024 times the total number of sub-conditions.
 
+    cost = (sum of largest F.threshold subcondition.cost values) + 1024 * F.subconditions.count
 
-### Fulfillment {#threshold-sha-256-condition-type-fulfillment}
+For example, if a threshold crypto-condition contains 5 sub-conditions with costs of 64, 64, 82, 84 and 84 and has a threshold of 3, the cost is equal to the sum of the largest three sub-condition costs (82 + 84 + 84 = 250) plus 1024 times the number of sub-conditions (1024 * 5 = 5120): 5370
 
-    ThresholdSha256FulfillmentPayload ::= SEQUENCE {
-      threshold INTEGER (0..4294967295),
-      subfulfillments SEQUENCE OF ThresholdSubfulfillment
+### ASN.1 {#threshold-sha-256-condition-asn1}
+
+    -- Condition Fingerprint
+    ThresholdFingerprintContents ::= SEQUENCE {
+      threshold            INTEGER (1..65535),
+      subconditions        SET OF Condition
     }
 
-    ThresholdSubfulfillment ::= SEQUENCE {
-      weight INTEGER (0..4294967295) DEFAULT 1,
-      condition Condition OPTIONAL,
-      fulfillment Fulfillment OPTIONAL
+    -- Fulfillment 
+    ThresholdFulfillment ::= SEQUENCE {
+      subfulfillments      SEQUENCE OF Fulfillment,
```

Comment:
Yeah I couldn't find anything on it either, not an explicit mention of either. I guess I just assumed it, because that's how SETs are interpreted in a lot of programming languages. I agree that it's misleading.